src/middleware/isAuth.js

```diff
@@ -17,6 +17,12 @@ async function isAuth(req, res, next) {
           username: req.session.passport.user.username,
         });
         break;
+      case "google":
```

Comment:
The authorizeCondition here looks the same to the one from the case "github" above, so you could just group 2 cases for the same condition and remove the duplicated code:
```suggestion
      case "github":
      case "google":
          authorizeCondition = await Users.findOne({
              userId: req.session.passport.user.id,
              username: req.session.passport.user.username,
        });
        break;
      
```